### ISSUE TITLE
Feature request: Ability to set Binding (x/30/0) attribute from dashboard

### ISSUE THREAD

**Issue by sgryphon** [matter-js/python-matter-server#815]
In the dashboard UI, as well as exploring nodes and viewing (reading) attribute values, it would be good if there was a way to write attribute values, specifically to write the Binding attribute.

This could be hidden behind an Advanced mode (like Home Assistant administration) if there was worry that it could mess things up.

Example:

node_id = 6
attribute_path = "1/30/0"
binding target:
   node = 12
   group = null
   endpoint = 1
   cluster = 0x0006 (On/Off cluster -- needs to be a client cluster)
   fabricIndex = 2

We know the node and endpoint, and the relevant cluster. The valid binding target struct would depend on what clusters are in the descriptor client list ("1/29/2"), and the valid nodes and endpoints in the fabric (and the server list of the target node/endpoint)

Binding is a key feature for disconnected operation, e.g. where a switch can directly interface with a device (such as a light) by binding an On/Off client cluster with On/Off server on the destination light.

I have a Zemismart switch that has 0x6 client cluster supported, and I would like to bind it to a light (which all have 0x6 server).

![image](https://github.com/user-attachments/assets/92b8169b-b5e5-4588-8a4d-a0967fbe4393)

The device supports the binding attribute, but there is no way to set it in the dashboard (I can via the programming UI)

![image](https://github.com/user-attachments/assets/2636d20c-6ca2-4883-8137-a586a10f86c3)

**Comment by marcelveldt** [comment]
Yes, this is all on the long list - not a priority for now due to a lack of capacity

**Comment by lboue** [comment]
@sgryphon
Could you share the commands used to set up binding on the command line?

**Comment by sgryphon** [comment]
BTW @lboue what is that UI you have screen shots of?  I am using the Matter Server via Home Assistant, and some of the buttons look the same (e.g. the Interview, Update, etc along the bottom), but I don't have the same config screens (can only view nodes).

**Comment by lboue** [comment]
@sgryphon The first screenshot is [ZAP tool](https://github.com/project-chip/zap/releases) used with the CHIP SDK. The second one is from Python Matter Server WebUI. You can to enable the `Show in sidebar` checkbox in the `Matter Server addon`.

**Comment by Apollon77** [comment]
Ok, maybe also my 5cents about bindings :-)

As summarized above:
* you need a source device which supports the Binding cluster at all
* that device also needs to have a relevant client cluster defined for the type you want to use (e.g. OnOff)
* and ... the device also need to do anything with it ... but n fact you will not know for sure what exactly happens :-)
* Then you can add a proper entry in the binding table of the binding to configure the "target" for the action (as shown above)
* Then you need to add an ACL entry on the target node to allow the source node to access the relevant endpoint (that was missing above)

The reality is that there are not that many real devices out there that bring client clusters with them ... and even if they are there it is unclear if they are functual ... Because of the fact that currently no ecosystem out there supports setting that up I think it was also not the highest prio for device manufacturers.

**Comment by lboue** [comment]
@Apollon77 I'm going to use 2 development kits to do this binding test.

**Comment by hoppel118** [comment]
@Apollon77 
> and even if they are there it is unclear if they are functual ... Because of the fact that currently no ecosystem out there supports setting that up I think it was also not the highest prio for device manufacturers.

EVE launched the EVE Thermo Control some days ago. With that device you can control the temperature of up to 5 EVE Thermos. It also acts as a temperature sensor for the paired EVE Thermos.

It uses Matter bindings. You can set that up via the EVE app. So, it’s already possible, when you are an Apple Home user. 😉

**Comment by Apollon77** [comment]
@hoppel118 Yes and that's the only official device I know where bindings are communicated as a feature. :-) An other device I know are Innovelli Matter switches. But that's it what I know is out there. There could be more, but when well hidden.

**Comment by lboue** [comment]
I also noticed thatEve Motion 20EBY9901is able to trigger OnOff (6) cluster:

![image](https://github.com/user-attachments/assets/a8123e95-7fc2-4a66-be65-7eda547d48fe)

**Comment by lboue** [comment]
It was not easy but it finally works using a Websocket client and python-matter-server like @sgryphon did with PowerShell.
I'm in the process of documenting this.

I open a discussion here:
https://github.com/home-assistant-libs/python-matter-server/discussions/978

**Comment by sgryphon** [comment]
> no ecosystem out there supports setting that up I think it was also not the highest prio

So, classic chicken-and-egg issue ... no controllers support it, so no devices bother to support it, so no controllers bother to support it, etc.

Good to see Eve is using it internally for their devices, which gives a reason for generic controllers to add support, which might encourage more devices.

I think the local direct peer-to-peer function is one of the strengths of Matter compared to other systems, i.e. without needing the controller to be online, so good to see it starting to mature.

Also, thanks for pointing out the permission issue. I'll have to check if I can get the Zemismart working once I add the permissions.

**Comment by lboue** [comment]
I can confirm that Eve Thermo binding with a regular Matter Temperature sensor is working fine:

**Eve Thermo**
![image](https://github.com/user-attachments/assets/13c151fc-bd43-4199-867c-921becaaaa3f)

![image](https://github.com/user-attachments/assets/81b48a25-6ed3-4f19-a1d5-ee8b307fbd7f)

**Temperature sensor**
![image](https://github.com/user-attachments/assets/a32fbf33-b068-4ee3-b547-250a5d84bb29)
